### PR TITLE
[Model Compression / TensorFlow] Support exporting pruned model

### PR DESCRIPTION
**Pull request by liuzhe-lz**

### CODE REVIEW

File: nni/compression/tensorflow/compressor.py [hunk 1]
```diff
@@ -167,6 +211,43 @@ def compress(self):
...
+            Because Keras cannot save tensors without a ``Model``,
+            this will create a model, set all masks as its weights, and then save that model.
+            Masks in saved model will be named by corresponding layer name in compressed model.
+
+        Returns
+        -------
+        None
+        """
+        _logger.info('Saving model to %s', model_path)
+        input_shape = self.compressed_model._build_input_shape  # cannot find a public API
```
Comment (QuanluZhang) [inline nni/compression/tensorflow/compressor.py:234]: where is `_build_input_shape` assigned?

File: nni/compression/tensorflow/compressor.py [hunk 2]
```diff
@@ -167,6 +211,43 @@ def compress(self):
...
+        Parameters
+        ----------
+        model_path : path-like
+            The path passed to ``Model.save()``.
+            You can use ".h5" extension name to export HDF5 format.
+        mask_path : path-like or None
+            Export masks to the path when set.
+            Because Keras cannot save tensors without a ``Model``,
+            this will create a model, set all masks as its weights, and then save that model.
+            Masks in saved model will be named by corresponding layer name in compressed model.
```
Comment (QuanluZhang) [inline nni/compression/tensorflow/compressor.py:227]: so even the mask file has different format between pytorch and tensorflow?...

**Review by QuanluZhang — APPROVED** [review]

File: nni/compression/tensorflow/compressor.py [hunk 3]
```diff
@@ -233,6 +312,10 @@ def __init__(self, layer, config, pruner):
         _logger.info('Layer detected to compress: %s', self.layer.name)
 
     def call(self, *inputs):
+        self._update_weights()
+        return self.layer(*inputs)
+
+    def _update_weights(self):
```
Comment (J-shang) [inline nni/compression/tensorflow/compressor.py:318]: why we split this function?

**Review by J-shang — APPROVED** [review]

File: nni/compression/tensorflow/compressor.py (hunk 1 above)
Comment (liuzhe-lz) [reply nni/compression/tensorflow/compressor.py:234]: This is tensorflow's private API.

File: nni/compression/tensorflow/compressor.py (hunk 2 above)
Comment (liuzhe-lz) [reply nni/compression/tensorflow/compressor.py:227]: PyTorch masks are exported as PyTorch tensors. So there's no way to make TF masks compatible.

File: nni/compression/tensorflow/compressor.py (hunk 3 above)
Comment (liuzhe-lz) [reply nni/compression/tensorflow/compressor.py:318]: Because forward should not be the only allowed place to update weights.
